media/transcoder/src/test/java/com/example/transcoder/CreateJobFromPresetBatchModeTest.java

```diff
@@ -0,0 +1,158 @@
+  private static final String TEST_FILE_NAME = "ChromeCast.mp4";
+  private static final String TEST_FILE_PATH =
+      "src/test/java/com/example/transcoder/testdata/" + TEST_FILE_NAME;
+  private static final String INPUT_URI = "gs://" + BUCKET_NAME + "/" + TEST_FILE_NAME;
+  private static final String OUTPUT_URI_FOR_PRESET =
+      "gs://" + BUCKET_NAME + "/test-output-preset-batch-mode/";
+  private static final String PRESET = "preset/web-hd";
+  private static String PROJECT_ID;
+  private static String PROJECT_NUMBER;
+  private static String JOB_ID;
+  private static PrintStream originalOut;
+  private ByteArrayOutputStream bout;
+
+  private static String requireEnvVar(String varName) {
+    String varValue = System.getenv(varName);
+    assertNotNull(
+        String.format("Environment variable '%s' is required to perform these tests.", varName));
+    return varValue;
+  }
+
+  private static void deleteBucket(String bucketName) {
+    Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+    Bucket bucket = storage.get(bucketName);
+    if (bucket != null) {
+      Page<Blob> blobs = bucket.list();
+
+      for (Blob blob : blobs.iterateAll()) {
+        System.out.println(blob.getName());
+        storage.delete(bucketName, blob.getName());
+      }
+      bucket.delete();
+    }
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    PROJECT_ID = requireEnvVar("GOOGLE_CLOUD_PROJECT");
+    PROJECT_NUMBER = requireEnvVar("TRANSCODER_PROJECT_NUMBER");
+  }
+
+  @Before
+  public void beforeTest() throws IOException {
+    originalOut = System.out;
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+
+    deleteBucket(BUCKET_NAME);
+    Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+    storage.create(
+        BucketInfo.newBuilder(BUCKET_NAME)
+            .setStorageClass(StorageClass.STANDARD)
+            .setLocation(LOCATION)
+            .build());
+
+    BlobId blobId = BlobId.of(BUCKET_NAME, TEST_FILE_NAME);
+    BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
+    Path path = Paths.get(TEST_FILE_PATH);
+    storage.create(blobInfo, Files.readAllBytes(path));
+    bout.reset();
+  }
+
+  @Test
+  public void test_CreateJobFromPresetBatchMode() throws Exception {
+    String jobName = String.format("projects/%s/locations/%s/jobs/", PROJECT_NUMBER, LOCATION);
+    CreateJobFromPresetBatchMode.createJobFromPresetBatchMode(
+        PROJECT_ID, LOCATION, INPUT_URI, OUTPUT_URI_FOR_PRESET, PRESET);
+    String output = bout.toString();
+    assertThat(output, containsString(jobName));
+    String[] arr = output.split("/");
+    JOB_ID = arr[arr.length - 1].replace("\n", "");
+
+    for (int attempt = 0; attempt < 5; attempt++) {
```

Comment:
Instead of using for loop, is there a way to get the job status from operation id or the returned response (during job creation)?

Reply:
These samples do not return a value. I believe that was the style when they were created. I'd like to keep them consistent.